examples/ogbn_train_cugraph.py

```diff
@@ -0,0 +1,289 @@
+    print('The root is: ', root)
+    dataset = PygNodePropPredDataset(name=args.dataset, root=root)
+    split_idx = dataset.get_idx_split()
+
+    data = dataset[0]
+    if not args.use_directed_graph:
+        data.edge_index = to_undirected(data.edge_index, reduce="mean")
+
+    graph_store = cugraph_pyg.data.GraphStore()
+    graph_store[dict(
+        edge_type=('node', 'rel', 'node'),
+        layout='coo',
+        is_sorted=False,
+        size=(data.num_nodes, data.num_nodes),
+    )] = data.edge_index
+
+    feature_store = cugraph_pyg.data.TensorDictFeatureStore()
+    feature_store['node', 'x'] = data.x
+    feature_store['node', 'y'] = data.y
+
+    data = (feature_store, graph_store)
+
+    if args.model == "GAT":
+        print(f"Training {args.dataset} with GAT model.")
+        model = torch_geometric.nn.models.GAT(
+            dataset.num_features, args.hidden_channels, args.num_layers,
+            dataset.num_classes, heads=args.num_gat_conv_heads).cuda()
+    elif args.model == "GCN":
+        print(f"Training {args.dataset} with GCN model.")
+        model = torch_geometric.nn.models.GCN(
+            dataset.num_features,
+            args.hidden_channels,
+            args.num_layers,
+            dataset.num_classes,
+        ).cuda()
+    elif args.model == "SAGE":
+        print(f"Training {args.dataset} with GraphSAGE model.")
+        model = torch_geometric.nn.models.GraphSAGE(
+            dataset.num_features,
+            args.hidden_channels,
+            args.num_layers,
+            dataset.num_classes,
+        ).cuda()
+    else:
+        raise ValueError('Unsupported model type: {args.model}')
+
+    optimizer = torch.optim.Adam(model.parameters(), lr=args.lr,
+                                 weight_decay=args.wd)
+
+    with tempfile.TemporaryDirectory(dir=args.tempdir_root) as samples_dir:
+        loader_kwargs = dict(
+            data=data,
+            num_neighbors=[args.fan_out] * args.num_layers,
+            replace=False,
+            batch_size=args.batch_size,
+            samples_dir=samples_dir,
+        )
+
+        train_loader = create_loader(
+            input_nodes=split_idx['train'],
+            stage_name='train',
+            shuffle=True,
+            **loader_kwargs,
+        )
+
+        val_loader = create_loader(
+            input_nodes=split_idx['valid'],
+            stage_name='val',
+            **loader_kwargs,
+        )
+
+        test_loader = create_loader(
+            input_nodes=split_idx['test'],
+            stage_name='test',
+            **loader_kwargs,
+        )
+        prep_time = round(time.perf_counter() - wall_clock_start, 2)
+        print("Total time before training begins (prep_time) =", prep_time,
+              "seconds")
+        print("Beginning training...")
+        val_accs = []
+        times = []
+        train_times = []
+        inference_times = []
+        best_val = 0.
+        start = time.time()
```

Comment:
Can you use `time.perf_counter()` for consistency across scripts in this PR?
```suggestion
        start = time.perf_counter()
```